src/names.c

```diff
@@ -213,22 +213,9 @@ static bool is_dotdotint(const char* name) {
   return (bool) strtol(name, NULL, 10);
 }
 
-static bool is_digit(const char c) {
-  switch (c) {
-  case '0':
-  case '1':
-  case '2':
-  case '3':
-  case '4':
-  case '5':
-  case '6':
-  case '7':
-  case '8':
-  case '9':
-    return true;
-  default:
-    return false;
-  }
+// Wrap for casting to unsigned char
+static bool is_digit(const unsigned char c) {
```

Comment:
The signedness of `char` is implementation defined. `isdigit()` takes an `int` (presumably so that `EOF` can be passed). It seems this is doing an unnecessary roundtrip of signedness casting here. As long as we pass `char`, this should be fine.

Reply:
It's convoluted, but https://stackoverflow.com/a/45007070/946850 suggests it's important.

Reply:
Interesting, the important bit seems to be:

> The problem is not just a theoretical one: several widespread C library implementations use the provided value straight as an index into a lookup table, so negative values will read into unallocated memory and segfault your program.

Should we rather use `unsigned char` everywhere we need bit data rather than integers?

Reply:
`isdigit()` is for character data, they come as `char*` from `CHAR()`. Other than that, perhaps `typedef unsigned char byte` would be useful.

Reply:
We could define `UCHAR()` and `UCHAR_RO()` as well.